contracts/reputationMiningCycle/ReputationMiningCycleCommon.sol

```diff
@@ -0,0 +1,99 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental "ABIEncoderV2";
+
+import "./../../lib/dappsys/math.sol";
+import "./../patriciaTree/PatriciaTreeProofs.sol";
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./ReputationMiningCycleStorage.sol";
+
+
+contract ReputationMiningCycleCommon is ReputationMiningCycleStorage, PatriciaTreeProofs, DSMath {
+  /// @notice Minimum reputation mining stake in CLNY
+  uint256 constant MIN_STAKE = 2000 * WAD;
+
+  function rewardResponder(address _responder) internal returns (bytes32) {
+    respondedToChallenge[_responder] = true;
+    uint256 reward = disputeRewardSize();
+    IColonyNetwork(colonyNetworkAddress).reward(_responder, reward);
+    rewardsPaidOut += reward;
+  }
+
+  function disputeRewardSize() internal view returns (uint256) {
+    // TODO: Is this worth calculating once, and then saving? Seems quite likely.
```

Comment:
It couldn't be on initialisation, it'd have to be the first time this was called during a dispute resolution (because we don't know how big the reward is going to be until the window is closed, and we know how many submissions there are).

Reply:
See comment above, I like the idea of calculating this once and saving it to a storage variable once the submission window is closed.